frontend/docsettings.lua

```diff
@@ -592,4 +609,65 @@ function DocSettings:flushCustomMetadata(doc_path)
     end
 end
 
+-- hash-based SDR storage
+
+local function getSdrsInDir(path)
+    -- Get all the metadata.filetype.lua files under directory path.
+    -- Derived from readerdictionary.getIfosInDir()
+    local sdrs = {}
+    local ok, iter, dir_obj = pcall(lfs.dir, path)
+    if ok then
+        for name in iter, dir_obj do
+            if name ~= "." and name ~= ".." then
+                local fullpath = path.."/"..name
+                local attributes = lfs.attributes(fullpath)
+                if attributes ~= nil then
+                    if attributes.mode == "directory" then
+                        local dirifos = getSdrsInDir(fullpath) -- recurse
+                        for _, ifo in pairs(dirifos) do
+                            table.insert(sdrs, ifo)
+                        end
+                    elseif fullpath:match("metadata%..+%.lua$") then
+                        table.insert(sdrs, fullpath)
+                    end
+                end
+            end
+        end
+    end
+    return sdrs
+end
+
+
+function DocSettings.getHashDirSdrInfos()
+   local sdrs = getSdrsInDir(DOCSETTINGS_HASH_DIR)
+   local title_author_strs = {}
+   for _, sdr in ipairs(sdrs) do
+    -- Ignore empty files
+    if lfs.attributes(sdr, "size") > 0 then
```

Comment:
That comment was copied from a few lines of code that I used for this (https://github.com/ryanwwest/koreader/blob/a43ec12b760c31372adfcce768e44bf1badb962b/frontend/docsettings.lua#L260-L264), which ignores files that are zero bytes. This does the same but since that would still register as a file and cause the file browsing slowness edge-case I wanted to list it.

Reply:
Oh ok.
This must date from the time we had crash and didn't fsync() docsettings when writting them, resulting in 0-bytes files...
Dunno if we can still get that (because of this test, we don't know about them :)
@NiLuJe: do you feel it is still needed ? (although it costs nothing to keep that code).

Reply:
I don't rightly recall why that is, but it vaguely rings a bell, so, yeah, I'd keep it in ;).

(Probably FUSE or macOS resource forks shenanigans or something).